SubPub/Tests/SubPubTests/SubPubTests.swift

```diff
@@ -0,0 +1,127 @@
+import XCTest
+@testable import SubPub
+
+final class SubPubTests: XCTestCase {
+    static var allTests = [
+        ("testSubResponse", testSubResponse),
+        ("testMultipleSubscribers", testMultipleSubscribers),
+        ("testUnsubscribing", testUnsubscribing),
+    ]
+    func testSubResponse() {
+        let manager = TestManager()
+        manager.start(with: self)
+        let subscriber = TestSubscriber(manager: manager)
+        // Test that subsciber state has updated from publication state
+        var publishedData = [TestData]()
+        switch manager.publisher.state {
+        case .loaded(let PubData):
+            publishedData = PubData
+        default:
+            break
+        }
+        
+        switch subscriber.state {
+        case .loaded(let subData):
+            XCTAssertEqual(subData, publishedData)
+        default:
+            XCTFail()
+        }
+    }
+    
+    func testMultipleSubscribers() {
+        let manager = TestManager()
+        manager.start(with: self)
+        
+        let subscriberA = TestSubscriber(manager: manager)
+        let subscriberB = TestSubscriber(manager: manager)
+        
+        // Test that both subscribers get new publication
+        let publishedError: TestError = .serviceError
+        manager.testError(publishedError)
+        
+        switch subscriberA.state {
+        case .error(let subError):
+            if let subTestError = subError as? TestError {
+                XCTAssertEqual(subTestError, publishedError)
+            }
+            else {
+                XCTFail()
```

Comment:
Generally it's good to include a message here

Reply:
yeah, good call. I was half expecting you to say something about having multiple asserts within a single test (which I'm on the fence about); thinking about refactoring...